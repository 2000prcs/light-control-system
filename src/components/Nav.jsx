import React, { Component } from 'react';

// const moment = require('moment');

export default class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date: 'Sunday 15 July, 2018',
      time: '8:13 PM',
      username: 'Moyeong Lee',
      initial: 'ML',
    };
  }

  componentDidMount() {
    this.setState({ time: this.getTime() });
    this.setState({ date: this.getDate() });
  	setInterval(() => {
      this.setState({ time: this.getTime() });
    }, 60000);
  }

  getDate() {
    const date = new Date().toString().slice(0, 15);
    return date;
  }

  getTime() {
    const time = `${new Date().toLocaleString().slice(11, 16)} ${new Date().toLocaleString().slice(20)}`;
    return time;
  }


  render() {
    return (
      <div className="nav">
        <div>
          {this.state.date}
        </div>
        <div>
          {this.state.time}
        </div>
        <div className="user">
          <div>
            {this.state.username}
          </div>
          <div>
            <svg width="50" height="50" >
                  <circle cx="25" cy="25" r="15" fill="#aeaeae" />
                  <text x="50%" y="50%" textAnchor="middle" fill="white" fontSize="15px" fontFamily="Arial" dy=".3em">{this.state.initial}</text>
                  Sorry, your browser does not support inline SVG.
            </svg>
          </div>
          <div>
            <i className="fas fa-sort-down"></i>
          </div>
        </div>
      </div>
    );
  }
}
