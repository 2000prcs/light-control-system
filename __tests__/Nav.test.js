import React from 'react';
import Nav from '../src/components/Nav.jsx';

const userData = {
  userName: 'Moyeong Lee',
};

describe('Nav component', () => {
  test('Should render Nav', () => {
    const NavWrapper = shallow(<Nav userData={userData}/>);
    expect(NavWrapper).toMatchSnapshot();
  });
});
