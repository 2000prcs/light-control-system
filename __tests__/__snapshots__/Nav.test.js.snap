// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component Should render Nav 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    userData={
      Object {
        "userName": "Moyeong Lee",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Sun Jul 22 2018
        </div>,
        <div>
          23:42:39
        </div>,
        <div
          className="user"
        >
          <div>
            Moyeong Lee
          </div>
          <div>
            <svg
              height="50"
              width="50"
            >
              <circle
                cx="25"
                cy="25"
                fill="#484848"
                r="15"
              />
              <text
                dy=".3em"
                fill="white"
                fontFamily="Arial"
                fontSize="15px"
                textAnchor="middle"
                x="50%"
                y="50%"
              >
                ML
              </text>
              Sorry, your browser does not support inline SVG.
            </svg>
          </div>
          <div>
            <i
              className="fas fa-sort-down"
            />
          </div>
        </div>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sun Jul 22 2018",
        },
        "ref": null,
        "rendered": "Sun Jul 22 2018",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "23:42:39",
        },
        "ref": null,
        "rendered": "23:42:39",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Moyeong Lee
            </div>,
            <div>
              <svg
                height="50"
                width="50"
              >
                <circle
                  cx="25"
                  cy="25"
                  fill="#484848"
                  r="15"
                />
                <text
                  dy=".3em"
                  fill="white"
                  fontFamily="Arial"
                  fontSize="15px"
                  textAnchor="middle"
                  x="50%"
                  y="50%"
                >
                  ML
                </text>
                Sorry, your browser does not support inline SVG.
              </svg>
            </div>,
            <div>
              <i
                className="fas fa-sort-down"
              />
            </div>,
          ],
          "className": "user",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Moyeong Lee",
            },
            "ref": null,
            "rendered": "Moyeong Lee",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                height="50"
                width="50"
              >
                <circle
                  cx="25"
                  cy="25"
                  fill="#484848"
                  r="15"
                />
                <text
                  dy=".3em"
                  fill="white"
                  fontFamily="Arial"
                  fontSize="15px"
                  textAnchor="middle"
                  x="50%"
                  y="50%"
                >
                  ML
                </text>
                Sorry, your browser does not support inline SVG.
              </svg>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <circle
                    cx="25"
                    cy="25"
                    fill="#484848"
                    r="15"
                  />,
                  <text
                    dy=".3em"
                    fill="white"
                    fontFamily="Arial"
                    fontSize="15px"
                    textAnchor="middle"
                    x="50%"
                    y="50%"
                  >
                    ML
                  </text>,
                  "Sorry, your browser does not support inline SVG.",
                ],
                "height": "50",
                "width": "50",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "cx": "25",
                    "cy": "25",
                    "fill": "#484848",
                    "r": "15",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "circle",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ML",
                    "dy": ".3em",
                    "fill": "white",
                    "fontFamily": "Arial",
                    "fontSize": "15px",
                    "textAnchor": "middle",
                    "x": "50%",
                    "y": "50%",
                  },
                  "ref": null,
                  "rendered": "ML",
                  "type": "text",
                },
                "Sorry, your browser does not support inline SVG.",
              ],
              "type": "svg",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-sort-down"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-sort-down",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Sun Jul 22 2018
          </div>,
          <div>
            23:42:39
          </div>,
          <div
            className="user"
          >
            <div>
              Moyeong Lee
            </div>
            <div>
              <svg
                height="50"
                width="50"
              >
                <circle
                  cx="25"
                  cy="25"
                  fill="#484848"
                  r="15"
                />
                <text
                  dy=".3em"
                  fill="white"
                  fontFamily="Arial"
                  fontSize="15px"
                  textAnchor="middle"
                  x="50%"
                  y="50%"
                >
                  ML
                </text>
                Sorry, your browser does not support inline SVG.
              </svg>
            </div>
            <div>
              <i
                className="fas fa-sort-down"
              />
            </div>
          </div>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sun Jul 22 2018",
          },
          "ref": null,
          "rendered": "Sun Jul 22 2018",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "23:42:39",
          },
          "ref": null,
          "rendered": "23:42:39",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Moyeong Lee
              </div>,
              <div>
                <svg
                  height="50"
                  width="50"
                >
                  <circle
                    cx="25"
                    cy="25"
                    fill="#484848"
                    r="15"
                  />
                  <text
                    dy=".3em"
                    fill="white"
                    fontFamily="Arial"
                    fontSize="15px"
                    textAnchor="middle"
                    x="50%"
                    y="50%"
                  >
                    ML
                  </text>
                  Sorry, your browser does not support inline SVG.
                </svg>
              </div>,
              <div>
                <i
                  className="fas fa-sort-down"
                />
              </div>,
            ],
            "className": "user",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Moyeong Lee",
              },
              "ref": null,
              "rendered": "Moyeong Lee",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <svg
                  height="50"
                  width="50"
                >
                  <circle
                    cx="25"
                    cy="25"
                    fill="#484848"
                    r="15"
                  />
                  <text
                    dy=".3em"
                    fill="white"
                    fontFamily="Arial"
                    fontSize="15px"
                    textAnchor="middle"
                    x="50%"
                    y="50%"
                  >
                    ML
                  </text>
                  Sorry, your browser does not support inline SVG.
                </svg>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <circle
                      cx="25"
                      cy="25"
                      fill="#484848"
                      r="15"
                    />,
                    <text
                      dy=".3em"
                      fill="white"
                      fontFamily="Arial"
                      fontSize="15px"
                      textAnchor="middle"
                      x="50%"
                      y="50%"
                    >
                      ML
                    </text>,
                    "Sorry, your browser does not support inline SVG.",
                  ],
                  "height": "50",
                  "width": "50",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "cx": "25",
                      "cy": "25",
                      "fill": "#484848",
                      "r": "15",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "circle",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ML",
                      "dy": ".3em",
                      "fill": "white",
                      "fontFamily": "Arial",
                      "fontSize": "15px",
                      "textAnchor": "middle",
                      "x": "50%",
                      "y": "50%",
                    },
                    "ref": null,
                    "rendered": "ML",
                    "type": "text",
                  },
                  "Sorry, your browser does not support inline SVG.",
                ],
                "type": "svg",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-sort-down"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-sort-down",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
